### PR TITLE
Fixes on headloss constraints

### PR DESCRIPTION
**Pull request by FJanssen-TNO**

### CODE REVIEW

**Comment by FJanssen-TNO** [comment]
@KobusVanRooyen ready for review. Bugfix in the setting the dH variable to __head_loss depending on flow direction.

File: src/mesido/head_loss_class.py [hunk 1]
```diff
@@ -1436,6 +1436,15 @@ def _pipe_head_loss_constraints(
...
             constraints.append(((dh - head_loss + flow_dir * big_m) / big_m, 0.0, np.inf))
 
+            constraints.append(
+                (
+                    (-dh - head_loss - (1 - flow_dir) * big_m) / big_m,
+                    -np.inf,
+                    0.0,
+                )
+            )
+            constraints.append(((dh - head_loss - flow_dir * big_m) / big_m, -np.inf, 0.0))
```
Comment (KobusVanRooyen) [inline src/mesido/head_loss_class.py:1446]: The current implementation looks like an inequality constraint <= 0.0 when flow_dir = 1. To make this an equality constraint for dh and head_loss taking into account flow_dir is the following constraint not maybe missing:
( (-dh - head_loss + (1 - flow_dir) * big_m) / ...., 0.0, np.inf )

**Review by KobusVanRooyen — COMMENTED** [review]
@FJanssen-TNO review has been completed

**Comment by KobusVanRooyen** [comment]
@FJanssen-TNO the review has been completed

File: src/mesido/head_loss_class.py (hunk 1 above)
Comment (FJanssen-TNO) [reply src/mesido/head_loss_class.py:1446]: That constraint is a few lines above 1430-1436

File: examples/pipe_diameter_sizing/src/example.py [hunk 2]
```diff
@@ -78,10 +78,10 @@ def energy_system_options(self):
     def pipe_classes(self, pipe):
         return [
             PipeClass("None", 0.0, 0.0, (0.0, 0.0), 0.0),
-            PipeClass("DN40", 0.0431, 1.5, (0.179091, 0.005049), 1.0),
-            PipeClass("DN50", 0.0545, 1.7, (0.201377, 0.006086), 2.0),
-            PipeClass("DN65", 0.0703, 1.9, (0.227114, 0.007300), 3.0),
-            PipeClass("DN80", 0.0825, 2.2, (0.238244, 0.007611), 4.0),
+            # PipeClass("DN40", 0.0431, 1.5, (0.179091, 0.005049), 1.0),
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply examples/pipe_diameter_sizing/src/example.py:81]: Scaling was giving issues with too many too small options.

[the remark this answers is not in the document]
Comment (KobusVanRooyen) [reply examples/pipe_diameter_sizing/src/example.py:81]: I suggest to either add comment on why the "# PipeClass("DN40", 0.0431, 1.5, (0.179091, 0.005049), 1.0)," is still in the code, and why it should not be deleted. Or rather just delete the line

[the remark this answers is not in the document]
Comment (KobusVanRooyen) [reply examples/pipe_diameter_sizing/src/example.py:84]: Comment has been added

**Review by KobusVanRooyen — APPROVED** [review]

**Comment by KobusVanRooyen** [comment]
@FJanssen-TNO approved

**Pull request merged**